ibis/backends/tests/test_register.py

```diff
@@ -0,0 +1,321 @@
+    if con.name != "datafusion":
+        table.count().execute()
+
+
+@pytest.mark.notimpl(["datafusion"])
+@pytest.mark.notyet(
+    [
+        "bigquery",
+        "clickhouse",
+        "dask",
+        "impala",
+        "mssql",
+        "mysql",
+        "pandas",
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+        "trino",
+    ]
+)
+def test_register_csv_gz(con, data_directory, gzip_csv):
+    with pushd(data_directory):
+        table = con.register(gzip_csv)
+
+    assert table.count().execute()
+
+
+@pytest.mark.notyet(
+    [
+        "bigquery",
+        "clickhouse",
+        "dask",
+        "impala",
+        "mssql",
+        "mysql",
+        "pandas",
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+        "trino",
+    ]
+)
+def test_register_with_dotted_name(con, data_directory, tmp_path):
+    basename = "foo.bar.baz/diamonds.csv"
+    f = tmp_path.joinpath(basename)
+    f.parent.mkdir()
+    data = data_directory.joinpath("diamonds.csv").read_bytes()
+    f.write_bytes(data)
+    table = con.register(str(f.absolute()))
+
+    if con.name != "datafusion":
+        table.count().execute()
+
+
+@pytest.mark.parametrize(
+    ("fname", "in_table_name", "out_table_name"),
+    [
+        pytest.param(
+            "parquet://functional_alltypes.parquet",
+            None,
+            "ibis_read_parquet",
+        ),
+        ("functional_alltypes.parquet", "funk_all", "funk_all"),
+        pytest.param("parquet://functional_alltypes.parq", "funk_all", "funk_all"),
+        ("parquet://functional_alltypes", None, "ibis_read_parquet"),
+    ],
+)
+@pytest.mark.notyet(
+    [
+        "bigquery",
+        "clickhouse",
+        "dask",
+        "impala",
+        "mssql",
+        "mysql",
+        "pandas",
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+        "trino",
+    ]
+)
+def test_register_parquet(
+    con, tmp_path, data_directory, fname, in_table_name, out_table_name
+):
+    fname = Path(fname)
+    _, table = next(read_tables([fname.stem], data_directory))
+
+    pq.write_table(table, tmp_path / fname.name)
+
+    with pushd(tmp_path):
+        table = con.register(f"parquet://{fname.name}", table_name=in_table_name)
+
+    assert any(t.startswith(out_table_name) for t in con.list_tables())
+
+    if con.name != "datafusion":
+        table.count().execute()
+
+
+@pytest.mark.notyet(
+    [
+        "bigquery",
+        "clickhouse",
+        "dask",
+        "datafusion",
+        "impala",
+        "mssql",
+        "mysql",
+        "pandas",
+        "polars",  # polars supports parquet dirs, not lists of files
+        "postgres",
+        "pyspark",
+        "snowflake",
+        "sqlite",
+        "trino",
+    ]
+)
+def test_register_iterator_parquet(
+    tmp_path,
+    data_directory,
+):
+    pq = pytest.importorskip("pyarrow.parquet")
```

Comment:
Hm, any idea why this code isn't hit? Since this line isn't ever hit, that means it's not the import failing, but something else that is causing this test to never execute.

Reply:
I forgot to add a `backend` or `con` fixture -- nice catch!